src/SmoFacade/BackupFileTools.cs

```diff
@@ -45,5 +47,23 @@ public static BackupType BackupTypeAbbrevToType(string type)
           throw new ArgumentException("Invalid backup type");
       }
     }
+
+    public static bool IsValidPath(string path)
+    {
+      // A quick check before leaning on exceptions
+      if(Path.GetInvalidPathChars().Any(path.Contains))
+        return false;
+
+      try {
+        // This will throw an argument exception if the path is invalid
+        Path.GetFullPath(path);
+        // A relative path won't help us much if the destination is another server. It needs to be rooted.
+        return Path.IsPathRooted(path) || path.StartsWith(@"\\");

```

Comment:
main change is that I added this here: `|| path.StartsWith(@"\\");`

The other stuff is just refactoring to make testing easier